python/mxnet/gluon/parameter.py

```diff
@@ -662,6 +662,7 @@ def cast(self, dtype):
             The new data type.
         """
         self._dtype = dtype
+        self._var = None  # Clear Symbol Variable as it caches the dtype
```

Comment:
how is this related to the cached graph and cached op fields?

Reply:
The _var is used as input